src/day07/CrabsPosition.php

```diff
@@ -0,0 +1,23 @@
+<?php declare(strict_types=1);
+
+namespace AdventOfCode\day07;
+
+final class CrabsPosition
+{
+    public function getLowestFuelSpend(array $input): int
+    {
+        $max = 0;
+        for ($a = 0; $a <= max($input); $a++) {
+            $sum = 0;
+            for ($x = 0; $x < count($input); $x++) {
+                for($s = 1; $s <= abs($input[$x]-$a); $s++) {
+                    $sum+=$s;
+                }
+            }
+            if($sum<$max || $max==0){
+                $max=$sum;
+            }
+        }
+        return $max;
```

Comment:
Jestli tenhle algoritmus vrací 168 a podle zadání má být 37, tak nepočítá správně. Nevím jak to bylo myšleno, ale podle mě tři zanořené cykly nejsou potřeba. Já to zkusil takhle a dostal jsem se k tomu výsledku 37.
```suggestion
        $minPos = min($input);
	$maxPos = max($input);
	$minFuelConsumption = count($input)*$maxPos; 
 
	for ($currPos = $minPos; $currPos <= $maxPos; $currPos++){	
	  $currFuelConsumption = 0;	  
	  
	  for ($i = 0; $i < count($input); $i++){	   
	   
	    $currFuelConsumption += abs($currPos-$input[$i]);	    
	    if ($currFuelConsumption > $minFuelConsumption){
	      break;
	    }
	  }	  
	  if ($currFuelConsumption < $minFuelConsumption){
	      $minFuelConsumption = $currFuelConsumption;
	  }
	}
	
	return $minFuelConsumption;
```

Reply:
> Jestli tenhle algoritmus vrací 168 a podle zadání má být 37, tak nepočítá správně. Nevím jak to bylo myšleno, ale podle mě tři zanořené cykly nejsou potřeba. Já to zkusil takhle a dostal jsem se k tomu výsledku 37. A ty indexy cyklů $a, $x, $s mi přijdou matoucí. Ve for cyklech se obvykle indexuje i,j,k... nebo se použije nějaký více vypovídající název proměnné.

I would agree with the $a, $x and $s are not ideal names for those "for" indexes.